### ISSUE TITLE
Tutorial template + gold standard example

### ISSUE THREAD

**Issue by shannonbux** [gatsbyjs/gatsby#6778]
Template for tutorial

## When should I write a tutorial?
A tutorial is a step-by-step process while a guide is more of a reference document (see [When Should I Write a Guide? here]())

Therefore, you should write a tutorial to help readers accomplish a task step-by-step.

## Tutorials are for everyone!
A common misconception is that tutorials are "just for beginners." While experts often don't need step-by-step tutorials and prefer reference documents (like an API reference doc), even experts have areas of knowledge they don't know yet. For example, GraphQL is new to many people who start building Gatsby sites. 

The core message here is to feel proud of writing tutorials because they could be useful to all sorts of people, and never feel embarrassed for wanting to explore tutorials! They are great!
 
## What topic should a tutorial cover?
Most people don't have the time or energy to focus on a tutorial that takes longer than 1-2 hours to complete. Try to choose a very specific task that someone could complete in that amount of time.

One Gatsby user said: "I like tutorials that take one cup of coffee. Some tutorials take too long, and require multiple cups of coffee."

So try to make a "one cup of coffee" tutorial.

Single-page tutorial: A great rule is to choose a task that someone can accomplish somewhere between 1-2 hours.

Multi-page tutorial: Each page should cover a task that someone can accomplish somewhere between 1-2 hours so they can finish the tutorial over several days.

TEMPLATE
## Goal of this tutorial?
Tell your readers what they can expect to accomplish by reading your tutorial. A great way to do this is a bulleted list of what they will have built by the end of the tutorial, e.g. At the end of this tutorial, you will have:
* built a Gatsby site
* deployed the Gatsby site
* shared your Gatsby site with friends

## Why complete this tutorial?
Here's your chance to tell your reader how their life is going to be more awesome after going through your tutorial. This can help them feel motivated to complete a difficult task.

## Table of Contents
If there are more than 3-5 headers, you ought to consider creating a table of contents to make the tutorial easier to scan

* Item One(#item-one)
* Item Two(#item-two)
* Item Three(#item-three)

## Step 1
Describe the first step a person ought to take to complete the task.

Show screenshots, GIFs, and/or code examples to help them. It might even help to share an example repo that they can clone or just peruse.

```shell
code example: make sure to use Prettier to format the code correctly!
```

Use introductions and captions for your screenshots and GIFs. This screenshot shows how to submit a pull request:

![screenshot of how to submit a pull request](submit-a-pull-request.png)

Caption: Screenshot of how to submit a pull request

## Step 2
[Same instructions as step 1]

## Step 3
[Same instructions as step 1]

## What did you accomplish in this tutorial?
List of things you accomplished
* thing 1
* thing 2
* thing 3

## Next step
Here is a great place to suggest more things they can do to keep learning about this topic:
* Look at these other tutorials: _______
* Look at the source code for more projects that exemplify the principles in this tutorial
* Share your project on socia media
* Look at these guides: ________

**Comment by shannonbux** [comment]
Need to get this reviewed and add gold standard example!

**Comment by shannonbux** [comment]
closed by #6489
